File: android/src/main/java/me/timeto/app/ui/home/bar/HomeBarTaskFolderButton.kt
package me.timeto.app.ui.home.bar

import androidx.compose.animation.core.Animatable
import androidx.compose.foundation.layout.size
import androidx.compose.material.Icon
import androidx.compose.runtime.Composable
import androidx.compose.runtime.LaunchedEffect
import androidx.compose.runtime.remember
import androidx.compose.ui.Modifier
import androidx.compose.ui.draw.scale
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.res.painterResource
import me.timeto.app.R
import me.timeto.app.ui.symbol.SymbolView
import me.timeto.shared.TaskFolderUi
import me.timeto.shared.vm.home.bar.HomeBarAnimate

@Composable
fun HomeBarTaskFolderButton(
    taskFolderUi: TaskFolderUi,
    color: Color,
    modifier: Modifier,
    onClick: () -> Unit,
) {

    val scaleAnimation = remember { Animatable(1f) }

    LaunchedEffect(Unit) {
        HomeBarAnimate.flow.collect { homeBarAnimate ->
            if ((homeBarAnimate is HomeBarAnimate.TaskFolder) &&
                (homeBarAnimate.taskFolderId == taskFolderUi.taskFolderDb.id)
            ) {
                homeBarButtonAnimate(scaleAnimation)
            }
        }
    }

    HomeBarIconButton(
        onClick = onClick,
        modifier = modifier,
    ) {
        if (taskFolderUi.taskFolderDb.isToday) {
            Icon(
                painterResource(id = R.drawable.ms_wb_sunny_fill),
                contentDescription = "Today",
                tint = color,
                modifier = Modifier
                    .size(homeBarIconSize)
                    .scale(scaleAnimation.value),
            )
        } else if (taskFolderUi.taskFolderDb.isTomorrow) {
            Icon(
                painterResource(id = R.drawable.ms_dark_mode_fill),
                contentDescription = "Tomorrow",
                tint = color,
                modifier = Modifier
                    .size(homeBarIconSize)
                    .scale(scaleAnimation.value),
            )
        } else {
            SymbolView(
                symbol = taskFolderUi.symbol,
                color = color,
                letterSize = homeBarLetterSize,
                iconSize = homeBarIconSize,
                emojiSize = homeBarLetterSize,
                modifier = Modifier
                    .scale(scaleAnimation.value),
            )
        }
    }
}

File: apple/iphone/UI/Home/Bar/HomeBarTaskFolderButton.swift
import SwiftUI
import shared

struct HomeBarTaskFolderButton: View {
    
    let taskFolderUi: TaskFolderUi
    let color: Color
    let onClick: () -> Void
    
    var body: some View {
        
        /*
        // todo
        val scaleAnimation = remember { Animatable(1f) }

        // todo
        LaunchedEffect(Unit) {
            homeTasksBarFolderAnimateFlow.collect { folderId ->
                if (folderId == taskFolderUi.taskFolderDb.id) {
                    scaleAnimation.animateTo(1.40f)
                    scaleAnimation.animateTo(1f)
                    scaleAnimation.animateTo(1.25f)
                    scaleAnimation.animateTo(1f)
                }
            }
        }
        */
        
        HomeBarIconButton(
            onClick: {
                onClick()
            },
            content: {
                ZStack {
                    if (taskFolderUi.taskFolderDb.isToday) {
                        Image(systemName: "sun.min.fill")
                            .font(.system(size: homeBarIconSize, weight: .semibold))
                            .foregroundColor(color)
                    } else if (taskFolderUi.taskFolderDb.isTomorrow) {
                        Image(systemName: "moon.fill")
                            .font(.system(size: homeBarIconSize, weight: .semibold))
                            .foregroundColor(color)
                    } else {
                        SymbolView(
                            symbol: taskFolderUi.symbol,
                            color: color,
                            letterSize: homeBarLetterSize,
                            iconSize: homeBarIconSize,
                            emojiSize: homeBarLetterSize,
                        )
                    }
                }
            },
        )
    }
}
